### PR TITLE
[bitnami/*] Add CD step to retrieve registry credentials

### PR DESCRIPTION
**Pull request by FraPazGal**
Signed-off-by: FraPazGal <fdepaz@vmware.com>

### Description of the change

This is a follow-up of https://github.com/bitnami/containers/pull/5477 to retrieve the registry credentials in each CD execution given the password's transient nature. We also decided to retrieve the `username` this way to avoid potential issues in the future.

### Benefits

CD pipelines failing with auth issues will now work as expected.

### Possible drawbacks

None

### Additional information

Tested in fork: https://github.com/FraPazGal/containers/runs/8245071416?check_suite_focus=true

### CODE REVIEW

**Review by beltran-rubo — DISMISSED** [review]

File: .github/workflows/cd-pipeline.yaml [hunk 1]
```diff
@@ -86,6 +86,17 @@ jobs:
...
+        name: Get staging marketplace's registry credentials
+        run: |
+          csp_auth_token=$(curl -s -H 'Content-Type: application/x-www-form-urlencoded' -X POST -d "grant_type=refresh_token&api_token=${{ secrets.STAGING_MARKETPLACE_API_TOKEN }}" https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/authorize | jq -r .access_token)
+          repo_info=$(curl -s -X POST -H "Content-Type: application/json" -H "csp-auth-token:$csp_auth_token" -d '{"withCredentials":true, "storageType":"OCI"}' https://gtwstg.market.csp.vmware.com/api/v1/repositories/transient)
+          marketplace_user=$(echo "$repo_info" | jq -r .response.repodetails.username)
+          marketplace_passwd=$(echo "$repo_info" | jq -r .response.repodetails.token)
+          echo "::add-mask::${marketplace_user}"
+          echo "::add-mask::${marketplace_passwd}"
+          echo "::set-output name=marketplace_user::${marketplace_user}"
+          echo "::set-output name=marketplace_passwd::${marketplace_passwd}"
```
Comment (aruiz14) [inline .github/workflows/cd-pipeline.yaml:99]: By any chance, are these values visible somewhere in the UI?

Comment (aruiz14) [reply .github/workflows/cd-pipeline.yaml:99]: Nevermind, I've just seen the `add-mask` part, which I assume is used for that purpose.

File: .github/workflows/cd-pipeline.yaml [hunk 2]
```diff
@@ -86,6 +86,17 @@ jobs:
             # Container folder doesn't exists we are assuming a deprecation
             echo "::set-output name=result::skip"
           fi
+      - id: get-registry-credentials
+        name: Get staging marketplace's registry credentials
+        run: |
+          csp_auth_token=$(curl -s -H 'Content-Type: application/x-www-form-urlencoded' -X POST -d "grant_type=refresh_token&api_token=${{ secrets.STAGING_MARKETPLACE_API_TOKEN }}" https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/authorize | jq -r .access_token)
```
Comment (aruiz14) [inline .github/workflows/cd-pipeline.yaml:92]:
```suggestion
          csp_auth_token=$(curl -s -H 'Content-Type: application/x-www-form-urlencoded' -X POST -d "grant_type=refresh_token&api_token=${{ secrets.STAGING_MARKETPLACE_API_TOKEN }}" https://console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/authorize | jq -re .access_token)
```

This way `jq` will fail if the key is not found

File: .github/workflows/cd-pipeline.yaml [hunk 3]
```diff
@@ -86,6 +86,17 @@ jobs:
...
+          repo_info=$(curl -s -X POST -H "Content-Type: application/json" -H "csp-auth-token:$csp_auth_token" -d '{"withCredentials":true, "storageType":"OCI"}' https://gtwstg.market.csp.vmware.com/api/v1/repositories/transient)
+          marketplace_user=$(echo "$repo_info" | jq -r .response.repodetails.username)
+          marketplace_passwd=$(echo "$repo_info" | jq -r .response.repodetails.token)
```
Comment (aruiz14) [inline .github/workflows/cd-pipeline.yaml:95]:
```suggestion
          marketplace_user=$(echo "$repo_info" | jq -re .response.repodetails.username)
          marketplace_passwd=$(echo "$repo_info" | jq -re .response.repodetails.token)
```

**Review by aruiz14 — DISMISSED** [review]
Approved with a couple minor suggestions

File: .github/workflows/cd-pipeline.yaml (hunk 1 above)
Comment (FraPazGal) [reply .github/workflows/cd-pipeline.yaml:99]: No, they aren't. By default, the `set-output` function does show the variable's value in the logs but the `add-mask` one makes the variable be treated as a secret (it shows *** in the UI). You can see an example of it in the [linked fork run](https://github.com/FraPazGal/containers/runs/8245071416?check_suite_focus=true):

```
::add-mask::***
::add-mask::***
::set-output name=marketplace_user::***
##[debug]steps.get-registry-credentials.outputs.marketplace_user='***'
::set-output name=marketplace_passwd::***
##[debug]steps.get-registry-credentials.outputs.marketplace_passwd='***'
##[debug]Finishing: Get staging marketplace's registry credentials
...
Substituting variable VIB_ENV_REGISTRY_URL in golang/vib-publish.json
Substituting variable VIB_ENV_REGISTRY_USERNAME in golang/vib-publish.json
Substituting variable VIB_ENV_REGISTRY_PASSWORD in golang/vib-publish.json
```

**Review by aruiz14 — APPROVED** [review]

**Pull request merged**